CMakeLists.txt

```diff
@@ -150,44 +150,53 @@ set(JS_DIRS
     services/user/CommonSys/ui:CommonSys_js
     services/user/TokenSys/ui:TokenSys_js
     services/user/ExploreSys/ui:ExploreSys_js
-    services/user/AdminSys/ui:AdminSys_js
     services/user/InviteSys/ui:InviteSys_js
 )
+set(ADMIN_DIR services/user/AdminSys/ui:AdminSys_js)
+set(COMMON_DIR services/user/CommonSys/common-lib/:CommonSysLib_js)
+
+function(add_js_target ENTRY OUTPUT_FILE)
+    string(REGEX REPLACE "^([^:]+):([^:]+)$" \\1 PATH ${ENTRY})
+    string(REGEX REPLACE "^([^:]+):([^:]+)$" \\2 DEP ${ENTRY})
+
+    file(GLOB_RECURSE sources ${CMAKE_CURRENT_SOURCE_DIR}/${PATH}/src/*)
+    set(DEPENDENCIES ${sources} ${ARGN})
+    add_custom_command(
+        OUTPUT ${CMAKE_CURRENT_SOURCE_DIR}/${PATH}/dist/${OUTPUT_FILE}
+        DEPENDS ${DEPENDENCIES}
+        COMMAND cd ${CMAKE_CURRENT_SOURCE_DIR}/${PATH} && yarn --mutex network && yarn build
+    )
+    add_custom_target(
+        ${DEP}
+        DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/${PATH}/dist/${OUTPUT_FILE}
+    )
+endfunction()
 
 option(BUILD_JS "Build JS" ON)
 if(BUILD_JS)
     list(TRANSFORM JS_DIRS REPLACE "^[^:]+:([^:]+)$" \\1 OUTPUT_VARIABLE JS_DEPS)
     file(GLOB common LIST_DIRECTORIES false ${CMAKE_CURRENT_SOURCE_DIR}/services/user/CommonSys/common/*)
     file(GLOB common-fonts LIST_DIRECTORIES false ${CMAKE_CURRENT_SOURCE_DIR}/services/user/CommonSys/common/fonts/*)
+    file(GLOB admin-wasm LIST_DIRECTORIES false ${CMAKE_CURRENT_SOURCE_DIR}/services/user/AdminSys/ui/wasm-psibase/*)
```

Comment:
This GLOB doesn't work correctly, because the files it's looking for are created by the build, which happens after cmake runs.

Reply:
I just deleted it because I think the dependency I was intending to add is already covered by `add_dependencies(AdminSys_js wasm-psibase)`

Reply:
I don't think it quite works.
- `wasm-pack` should run unconditionally because it manages its own dependencies ✅
- AdminSys_js should be built iff any of its sources were modified OR `wasm-pack` updated its outputs. As it's currently set up, there is nothing that detects whether `wasm-pack` did anything.  I believe the correct way to handle this is to use `BYPRODUCTS` in `add_custom_target`

Reply:
Added `BYPRODUCTS` and the corresponding dependencies